calendar-bundle/src/EventListener/BreadcrumbListener.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CalendarBundle\EventListener;
+
+use Contao\CalendarEventsModel;
+use Contao\CalendarModel;
+use Contao\Config;
+use Contao\CoreBundle\Framework\Adapter;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\Events;
+use Contao\Input;
+use Contao\PageModel;
+use Contao\StringUtil;
+
+class BreadcrumbListener
+{
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    public function __construct(ContaoFrameworkInterface $framework)
+    {
+        $this->framework = $framework;
+    }
+
+    public function onGenerateBreadcrumb(array $items): array
+    {
+        $eventAlias = $this->getEventAlias();
+        if (!$eventAlias) {
+            return $items;
+        }
+
+        $calendar = $this->getCalenderModel();
+        if (!$calendar) {
+            return $items;
+        }
+
+        $eventModel = $this->getEventModel($eventAlias, $calendar);
+        if (!$eventModel) {
+            return $items;
+        }
+
+        if ($GLOBALS['objPage']->requireItem) {
+            return $this->overrideActiveBreadcrumbItem($items, $eventModel);
+        }
+
+        return $this->addBreadcrumbItem($items, $eventModel);
+    }
+
+    private function getEventAlias(): ?string
+    {
+        if (!isset($GLOBALS['objPage'])) {
+            return null;
+        }
+
+        /** @var Adapter|Input $inputAdapter */
+        $inputAdapter = $this->framework->getAdapter(Input::class);
+        /** @var Adapter|Config $configAdapter */
+        $configAdapter = $this->framework->getAdapter(Config::class);
+
+        if ($configAdapter->get('useAutoItem')) {
+            return $inputAdapter->get('auto_item');
+        }
+
+        return $inputAdapter->get('items');
```

Comment:
This is wrong. The event reader is triggered by the `events` parameter.

Reply:
Copy & paste error. Fixed with 3f2f4aa31630189a8e1d5cc3ea92ff62f3072877.